src/common/cockpitlocale.c

```diff
@@ -64,6 +64,44 @@ cockpit_locale_from_language (const gchar *value,
   return result;
 }
 
+gchar *
+cockpit_language_from_locale (const gchar *value)
+{
+  gint len = -1;
+  const gchar *underscore;
+  const gchar *dot;
+  gchar *country = NULL;
+  gchar *lang = NULL;
+  gchar *result = NULL;
+
+  if (value == NULL || g_strcmp0 (value, "C") == 0)
+    return NULL;
+
+  dot = strchr (value, '.');
+  underscore = strchr (value, '_');
+
+  if (underscore)
+    {
+      if (dot)
+        len = (dot - underscore) - 1;
```

Comment:
This should get a check that `dot > underscore`. One expects that, but let's not crash the bridge or something else with malicious/broken locales.